app/main.py

```diff
@@ -1,2 +1,108 @@
+from dataclasses import dataclass
+from typing import Hashable, Any, Optional
+from fractions import Fraction
+
+
 class Dictionary:
-    pass
+    _INITIAL_CAPACITY = 8
+    _RESIZE_THRESHOLD = Fraction(2, 3)
+    _CAPACITY_MULTIPLIER = 2
+
+    @dataclass
+    class Node:
+        key: Hashable
+        value: Any
+        hash_value: int
+
+    def __init__(
+            self,
+            capacity: int = _INITIAL_CAPACITY
```

Comment:
I still don't agree that we should pass capacity as a parameter. Dict can take an iterable on init that will set values on creation, but not parameters that will affect its internal organization